prov/efa/src/rxr/rxr_atomic.c

```diff
@@ -38,6 +38,16 @@
 #include "rxr_atomic.h"
 #include "rxr_pkt_cmd.h"
 
+static void rxr_atomic_set_shm_addr(struct fi_msg_atomic *msg)
+{
+	int i;
+
+	if (!(shm_info->domain_attr->mr_mode & FI_MR_VIRT_ADDR)) {
+		for (i = 0; i < msg->rma_iov_count; i++)
+			*(uint64_t *)&msg->rma_iov[i].addr = 0;
```

Comment:
This line is quite confusing, should it just be ```msg->rma_iov[I].addr = 0```?

Reply:
This is for accessing the constant member `const struct fi_rma_iov *rma_iov` in struct  fi_msg_rma.

Reply:
If it is defined as constant, we should not forcefully change it. That might result in unexpected behavior from application. The application might reuse the rma_iov later and did not know the value was changed.

Reply:
The struct fi_msg_rma msg here in what we allocate internally, not the application.

Reply:
I know msg is internally allocated. It is ```shm_msg``` in the caller.  but ```shm_msg``` is a mem copy of ```msg```. Note ```msg->rma_iov``` is a pointer, not an array, See the following definition:

```
struct fi_msg_atomic {
	const struct fi_ioc *msg_iov; /* local scatter-gather array */
	void                **desc;   /* local access descriptors */
	size_t              iov_count;/* # elements in ioc */
	const void          *addr;    /* optional endpoint address */
	const struct fi_rma_ioc *rma_iov; /* remote SGL */
	size_t              rma_iov_count;/* # elements in remote SGL */
	enum fi_datatype    datatype; /* operand datatype */
	enum fi_op          op;       /* atomic operation */
	void                *context; /* user-defined context */
	uint64_t            data;     /* optional data */
};

```

so ```shm_msg->rma_iov``` and ```msg->rma_iov``` pointing to the same array that is allocated by caller, who does not expect it to be changed.

Reply:
I see your point. Updated accordingly.